### ISSUE TITLE
Generating the labels with torch.arange

### ISSUE THREAD

**Issue by ckusumadewa-lab** [MishaLaskin/curl#14]
First of all, thank you so much for kindly sharing your great research and also the code. 

However, I have one question regarding the labels generation from the logits using the following code (curl_sac.py line 424):

`labels = torch.arange(logits.shape[0]).long().to(self.device)`

What if, for example, we get several same observations in the batch sampled from the replay buffer? Isn't the code will set same features as different classes since we use `torch.arange`?

Please correct me if I am wrong. Thank you so much.

**Comment by MishaLaskin** [comment]
This happens sometimes, but the likelihood is low and InfoNCE loss is
robust to small perturbations since there are many other negatives. More
generally, this is an unsupervised learning loss, so we don't have access
to the image labels and therefore don't know if two observations are the
same (unless they coincide perfectly in pixel space, which is unlikely).

**Comment by aravindsrinivas** [comment]
Closing issue.
